### PR TITLE
chore: non-cached credentials provider

### PR DESCRIPTION
**Pull request by diegomarquezp**
Adds a lightweight version of `com.google.api.client.googleapis.auth.oauth2.DefaultCredentialProvider` that does not cache the default credentials (see [here](https://github.com/googleapis/google-api-java-client/blob/3cb09eb2dd5ed5f3eb5b9d18b7e4413d2f3634f1/google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java#L93-L100)). This allows the user to switch/revoke credentials in the gcloud CLI and get them automatically detected

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/NonCachedDefaultCredentialsProvider.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.AccessControlException;
+import java.util.Locale;
+
+/**
+ * Lightweight version of DefaultCredentialProvider
+ * @see <a href="https://github.dev/googleapis/google-api-java-client/blob/3cb09eb2dd5ed5f3eb5b9d18b7e4413d2f3634f1/google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java#L45">com.google.api.client.googleapis.auth.oauth2.DefaultCredentialProvider</a>
```
Comment (meltsufin) [inline plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/NonCachedDefaultCredentialsProvider.java:31]: Any reason you're not referencing the more modern `google-auth-library-java` [GoogleCredentials](https://github.com/googleapis/google-auth-library-java/blob/main/oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java)?

File: plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/NonCachedDefaultCredentialsProvider.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.AccessControlException;
+import java.util.Locale;
+
+/**
+ * Lightweight version of DefaultCredentialProvider
+ * @see <a href="https://github.dev/googleapis/google-api-java-client/blob/3cb09eb2dd5ed5f3eb5b9d18b7e4413d2f3634f1/google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java#L45">com.google.api.client.googleapis.auth.oauth2.DefaultCredentialProvider</a>  
+ */
+public class NonCachedDefaultCredentialsProvider {
```
Comment (meltsufin) [inline plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/NonCachedDefaultCredentialsProvider.java:33]: Copying the entire class just to make one small modification doesn't seem right. Can we consult with Auth team (@TimurSadykov), or file an FR in the [google-auth-library-java](https://github.com/googleapis/google-auth-library-java/issues) to add an option to clear the cache?

**Review by meltsufin — CHANGES_REQUESTED** [review]

File: plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/NonCachedDefaultCredentialsProvider.java (hunk 2 above)
Comment (diegomarquezp) [reply plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/NonCachedDefaultCredentialsProvider.java:33]: Filed a FR
https://github.com/googleapis/google-auth-library-java/issues/1181

**Comment by diegomarquezp** [comment]
replaced by https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/3731

**Pull request closed without merging**